third_party/libusb/webport/src/chrome_usb/chrome-login-state-hook.js

```diff
@@ -17,149 +17,100 @@
-goog.require('GoogleSmartCard.RequestReceiver');
-goog.require('goog.Promise');
 goog.require('goog.asserts');
-goog.require('goog.array');
-goog.require('goog.Disposable');
-goog.require('goog.iter');
 goog.require('goog.log');
-goog.require('goog.log.Logger');
-goog.require('goog.messaging.AbstractChannel');
-goog.require('goog.object');
-goog.require('goog.promise.Resolver');
 
 goog.scope(function() {
 
 const GSC = GoogleSmartCard;
 
-/**
- * This class implements a hook that hides USB devices when the app is running
- * for a user who has locked their screen. This causes PCSC Lite to release all
- * owned USB devices so that they can be accessed from the sign-in profile app.
- * @extends goog.Disposable
- * @constructor
- */
-GSC.Libusb.ChromeLoginStateHook = function() {
-  /**
-   * @type {boolean}
-   * @private
-   */
-  this.simulateDevicesAbsent_ = false;
-  /** @type {!Function} @private @const */
-  this.boundOnGotSessionState_ = this.onGotSessionState_.bind(this);
-  /** @type {!goog.promise.Resolver} @private @const */
-  this.hookIsReadyResolver_ = goog.Promise.withResolver();
-  if (chrome.loginState) {
-    chrome.loginState.getProfileType(this.onGotProfileType_.bind(this));
-  } else {
-    goog.log.warning(
-        this.logger,
-        'chrome.loginState API is not available. This app might require a ' +
-            'newer version of Chrome.');
-    this.hookIsReadyResolver_.reject();
-  }
-};
-
-const ChromeLoginStateHook = GSC.Libusb.ChromeLoginStateHook;
-
-goog.inherits(ChromeLoginStateHook, goog.Disposable);
-
-/**
- * @type {!goog.log.Logger}
- * @const
- */
-ChromeLoginStateHook.prototype.logger =
-    GSC.Logging.getScopedLogger('Libusb.LoginStateHook');
+const logger = GSC.Logging.getScopedLogger('LibusbLoginStateHook');
 
-/** @override */
-ChromeLoginStateHook.prototype.disposeInternal = function() {
-  if (chrome.loginState) {
-    chrome.loginState.onSessionStateChanged.removeListener(
-        this.boundOnGotSessionState_);
+// TODO(#429): Add event unsubscription.
+GSC.LibusbLoginStateHook = class extends GSC.LibusbProxyHook {
+  constructor() {
+    super();
+    /** @type {boolean} @private */
+    this.simulateDevicesAbsent_ = false;
+    /** @type {!Function} @private @const */
+    this.boundOnGotSessionState_ = this.onGotSessionState_.bind(this);
+    /** @type {!Function|null} @private */
+    this.resolveInitializationPromise_ = null;
+    /** @type {!Promise<void>} @private @const */
+    this.initializationPromise_ = new Promise((resolve, reject) => {
+      this.resolveInitializationPromise_ = resolve;
+    });
+    if (chrome && chrome.loginState) {
+      chrome.loginState.getProfileType(this.onGotProfileType_.bind(this));
+    } else {
+      goog.log.warning(
+          logger,
+          'chrome.loginState API is not available. This app might require a ' +
+              'newer version of Chrome.');
+      this.resolveInitializationPromise_();
```

Comment:
I may misunderstand this, but isn't resolveInitializationPromise_ null until initializationPromise_ is called/awaited, which probably hasn't happened here?